### ISSUE TITLE
Code results are not ideal

### ISSUE THREAD

**Issue by Onedimensionalinfinite** [shrey16/caustic_based_surface_generation#3]
Hello, I used your program to calculate the pictures in the paper, and found that the result is not ideal. Can the result be optimized?

**Comment by shrey16** [comment]
Definitely, as I mentioned I used a first order method to solve the PDE
whereas the paper uses a second order method, so you may want to try and
implement that.

**Comment by Onedimensionalinfinite** [comment]
Thank you very much for replying to my message, I am very excited. Looking at the code before, I didn't find that you were using the first-order method. The paper is a second-order PDE, why your code can solve this problem. I was wondering if the code for solving Poisson's equation could be independently developed, but I still don't have enough knowledge of Python to solve this problem.

**Comment by Onedimensionalinfinite** [comment]
Hello, does the first-order method you mentioned earlier refer to the first-order finite difference method?

**Comment by shrey16** [comment]
Yeah

**Comment by Onedimensionalinfinite** [comment]
Hello, I encountered a difficulty in the process of optimizing the finite difference method. What are the parameters of delta and w, ro in the poisson_solver function in the program?
